ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/router-bgp.j2

```diff
@@ -12,6 +12,34 @@ router bgp {{ router_bgp.as }}
 {%         endif %}
    {{ distance_cli }}
 {%     endif %}
+{# the ordering of graceful-restart commands in the CLI is as follow
+   graceful-restart restart-time 300
+   graceful-restart stalepath-time 42
+   graceful-restart
+
+   but when disabling graceful-restart using no graceful-restart, stalepath-time stays
+   hence the convoluted Jinja template
+#}
+{%     if router_bgp.graceful_restart.enabled is arista.avd.defined(true) and router_bgp.graceful_restart.restart_time is arista.avd.defined %}
+   graceful-restart restart-time {{ router_bgp.graceful_restart.restart_time }}
+{%     endif %}
+{%     if router_bgp.graceful_restart.stalepath_time is arista.avd.defined %}
+   graceful-restart stalepath-time {{ router_bgp.graceful_restart.stalepath_time }}
+{%     endif %}
+{%     if router_bgp.graceful_restart.enabled is arista.avd.defined(true) %}
+   graceful-restart
+{%     endif %}
+{%     if router_bgp.graceful_restart_helper.enabled is arista.avd.defined(false) %}
+   no graceful-restart-helper
+{%     elif router_bgp.graceful_restart_helper.enabled is arista.avd.defined(true) %}
+{# graceful-restart-help long-lived and restart-time are mutually exclusive in CLI #}
+{# restart-time will win in this case #}
+{%         if router_bgp.graceful_restart.restart_time is arista.avd.defined %}
```

Comment:
Shouldn't this look for graceful_restart_helper.restart time instead? Also below - graceful_restart doesn't have a long_lived key, the template should be looking for graceful_restart_helper.long_lived right?

Reply:
yeah - good point - you caught me in my copy-pasta fail! and because I have a test with everything on it worked..